### DISCUSSION TITLE
Report | Week 18

### DISCUSSION THREAD

**Discussion by sfrancisw** [SE-TINF22B6/time2shine#99] [Show and tell]
Hey there everyone, welcome to this weeks report of your team time2shine!

### Retro Summary
All in all our retro was very positive. There where some things that we would have done different or avoided completely, like the "not so optimized" game library we are using but other than that we discussed about a lot of positive things like our frontend and our teamwork.
We also discussed a lot of ideas for the future, like adding more games to our game hub, adding multiplayer and friendlists and so on. Unfortunately, the would consume too much time if we where to implement them now.
We talked about things we need to do for our demo as well. There is not much left though, mostly optimization, code cleanup and documentation.

---

### Development Updates

There has been a lot of improvements in the game section of our project. The snake game we've been working on is now fully functional and you can play it on our website. It is similar to the well known snake game but with a twist. There are Röhrigs walking around the field instead of apples. These enemies will split your snake, so you can control a maximum of 2 snakes at the same time with WASD and IJKL. You also get a score multiplicator while you have two snakes at the same time. Any additional snakes unfortunately will not survive. You'll have to try and stay alive for as long as possible before you get too small and die. 

![image](https://github.com/SE-TINF22B6/time2shine/assets/59262249/1809ac7e-0c77-4212-baa7-f70eccb9d41e)

We also added the new Snake Extreme game to our overview page, which shows all of our games. From there, you can start playing it to your heart's content. After you die ingame, your score is automatically submitted to our backend, where it is saved in the highscore database. If you did well enough, you can position yourself on our all-time leaderboard, so get ready to invest some hours into this new AAAA masterpiece. So head over to https://t2s.maiwald.cc/games and start playing right now!

We've also implemented the OpenCards API into our blackjack game, so you can now actually see what cards you are drawing. 
![grafik](https://github.com/SE-TINF22B6/time2shine/assets/49996344/c4692708-ea82-4315-98b2-e833c8ab68a3)

All in all the games are now in a very good state and we only have to make a few more tweaks for our demo next week.

**Comment by Elizabethstella** [comment]
Impressive progress by Team Time2Shine! Love the innovative snake game and enhanced blackjack. Excited to see the Demo!

**Comment by Stev3eM** [comment]
Well done so far. Interesting choice with splitting the snake and let the player control it both. I think one funczional game should be enough for this project. Looking forward to se the demo and your skills in the game :P
